chainlink-modsec/libmodsec/internal/executor/executor.go

```diff
@@ -0,0 +1,110 @@
+package executor
+
+import (
+	"fmt"
+	"github.com/smartcontractkit/chainlink-modsec/libmodsec/pkg/modsectypes"
+)
+
+type Executor struct {
+	stopCh    chan struct{}
+	messageCh chan modsectypes.Message
+	// List of sources which can potentially provide messages to be executed. We use an array here in case
+	// custom executors want to subscribe directly to a verifier storage
+	// Example could be a kafka topic, CL commit verifier, etc.
+	MessageSources []MessageReader
+
+	// List of sources which can potentially provide attestations for messages
+	// This could be changed to a map of verifierID to attestation reader
+	AttestationReader AttestationReader
+
+	// Map of destination chains to writers. A writer is some abstraction on the chain access layer
+	Transmitters map[uint64]modsectypes.ContractTransmitter
```

Comment:
Feels like its more complicated to make the executor multi-chain, I was thinking of the object as scoped to a single chain, could be wrong though.